src/shape.rs

```diff
@@ -660,237 +677,214 @@ impl ShapeLine {
                                 number_of_blanks += 1;
                             }
                             continue;
                         } else if wrap == Wrap::Glyph {
                             for (glyph_i, glyph) in word.glyphs.iter().enumerate().rev() {
-                                let glyph_size = font_size * glyph.x_advance;
-                                if fit_x - glyph_size >= 0. {
-                                    fit_x -= glyph_size;
-                                    word_range_width += glyph_size;
+                                let glyph_width = font_size * glyph.x_advance;
+                                if fit_x - glyph_width >= 0. {
+                                    fit_x -= glyph_width;
+                                    word_range_width += glyph_width;
                                     continue;
                                 } else {
-                                    word_ranges.push((
+                                    self.add_to_visual_line(
+                                        &mut current_visual_line,
+                                        span_index,
                                         (i, glyph_i + 1),
                                         fitting_start,
                                         word_range_width,
                                         number_of_blanks,
-                                    ));
+                                    );
+                                    visual_lines.push(current_visual_line);
+                                    current_visual_line = VisualLine::default();
+
                                     number_of_blanks = 0;
-                                    fit_x = line_width - glyph_size;
-                                    word_range_width = glyph_size;
+                                    fit_x = line_width - glyph_width;
+                                    word_range_width = glyph_width;
                                     fitting_start = (i, glyph_i + 1);
                                 }
                             }
                         } else {
                             // Wrap::Word
-                            let mut prev_word_width = None;
-                            if word.blank && number_of_blanks > 0 {
-                                // current word causing a wrap is a space so we ignore it
-                                number_of_blanks -= 1;
-                            } else if let Some(previous_word) = span.words.get(i - 1) {
-                                // Current word causing a wrap is not whitespace, so we ignore the
-                                // previous word if it's a whitespace
-                                if previous_word.blank {
-                                    number_of_blanks -= 1;
-                                    prev_word_width = Some(previous_word.x_advance * font_size);
+                            let mut trialing_space_width = None;
```

Comment:
This contains a typo: It should be renamed to `trailing_space_width`.

Reply:
Fixed in [7fa51c6](https://github.com/pop-os/cosmic-text/pull/108/commits/7fa51c6404f7f286a4c23abc79640bc0e63a5138)